src/dynamics/video_player/player/player.js

```diff
@@ -3100,13 +3100,17 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                         const {
                             selector,
                             id,
-                            adslotid
+                            adslotid,
+                            incontainer
                         } = location;
                         if (!selector || !(id || adslotid)) {
                             console.warn(`Please provide selector and adslotid for companion ad`);
                             return;
                         }
-                        const element = document.querySelector(selector);
+                        let element = document.querySelector(selector);
+                        if (!element && incontainer) {
+                            element = this.activeElement().querySelector(selector);
```

Comment:
Hi @wwongkargo for my opinion if `incontainer` is true look first or even only inside `activeElement`, if finding only inside is not critical try to search in whole document. As, there can be conflict on selectors and will find first outside of `activeElement` when we're trying to find something inside of it.

Reply:
Sounds good!